### PR TITLE
STM32: EXTI driver cleanup

### PR DESCRIPTION
**Pull request by erwango**
`driver/interrupt_controller/exti_stm32.c` holds a rather complex code which is hard to maintain and update.

Perform some clean up:
 - Extend IRQ table concept proposed for MP1 series
 - Remove unused (because non functional) code

### CODE REVIEW

**Comment by erwango** [comment]
@arnop2

**Review by arnopo — APPROVED** [review]

**Comment by erwango** [comment]
Set as DNM waiting for additional tests

**Comment by erwango** [comment]
Tested on all series.

File: drivers/interrupt_controller/exti_stm32.c [hunk 1]
```diff
@@ -51,7 +51,51 @@
 #define EXTI_LINES 49
 #endif
 
-#if defined(CONFIG_SOC_SERIES_STM32MP1X)
+#if defined(CONFIG_SOC_SERIES_STM32F0X) || \
+    defined(CONFIG_SOC_SERIES_STM32L0X)
+IRQn_Type exti_irq_table[] = {
```
Comment (mbolivar) [inline drivers/interrupt_controller/exti_stm32.c]: Adding const would save some RAM, no? Here and elsewhere.

**Review by mbolivar — APPROVED** [review]
Another nice cleanup!

File: drivers/interrupt_controller/exti_stm32.c (hunk 1 above)
Comment (erwango) [reply drivers/interrupt_controller/exti_stm32.c]: Sure, let me have a try. Thanks!

**Pull request merged**